dagshub/data_engine/model/datasource.py

```diff
@@ -1064,3 +1195,119 @@ def wrap_bytes(val: bytes) -> str:
 
     def get_metadata_entries(self):
         return self._metadata_entries
+
+
+@dataclass
+class DatasourceQuery(DataClassJsonMixin):
+    as_of: Optional[int] = field(default=None, metadata=config(exclude=exclude_if_none, letter_case=LetterCase.CAMEL))
+    select: Optional[List] = field(default=None, metadata=config(exclude=exclude_if_none))
```

Comment:
Don't want to define the List type?

Reply:
It's a list of dicts that aren't really structured.
I'll define that it's dictionaries inside at least